src/common/legacy/src/convert_function_to_cnn_network.cpp

```diff
@@ -2076,10 +2076,11 @@ void convertFunctionToICNNNetwork(const std::shared_ptr<const ::ngraph::Function
     auto params = graph->get_parameters();
     for ( const auto &param : params ) {
         const std::string input_name = param->get_friendly_name();
-        auto &thisInputData = *thisInputDataMap[input_name];
-        resultInputDataMap[input_name]->setPrecision(thisInputData.getPrecision());
-        resultInputDataMap[input_name]->setLayout(thisInputData.getLayout());
-        resultInputDataMap[input_name]->getPreProcess() = thisInputData.getPreProcess();
+        auto thisInputDataItr = thisInputDataMap.find(input_name);
+        IE_ASSERT(thisInputDataItr != thisInputDataMap.end());
+        resultInputDataMap[input_name]->setPrecision(thisInputDataItr->second->getPrecision());
+        resultInputDataMap[input_name]->setLayout(thisInputDataItr->second->getLayout());
+        resultInputDataMap[input_name]->getPreProcess() = thisInputDataItr->second->getPreProcess();
```

Comment:
Is this a fix to some existing test failure?
If so, could you explain wat was the problem and how this change is fixing it?

Reply:
This is fix for issues happening during conformance tests.
ASSERT does not resolve the issue it self it is to ensure issue will be spotted. It means error in earlier phases of the project

The real issue was in convert_function_to_cnn_network.cpp
It was changing name of input parameter of ngraph::Function.

Of course if there happen similar error (renaming of input for ngraph::Function) in non debug environment it will still segfault.
I can change 
```
IE_ASSERT(thisInputDataItr != thisInputDataMap.end());
```
to 
```
IE_ASSERT(thisInputDataItr != thisInputDataMap.end());
if(thisInputDataItr == thisInputDataMap.end())
{
    IE_THROW() << "Internal issue with model handling. Unpropper input names.";
}
```

Reply:
I am fine with:
if(thisInputDataItr == thisInputDataMap.end())
{
    IE_THROW() << "Internal issue with model handling. Unpropper input names.";
}

only I would append problematic input name to a message.

Reply:
Ok, I double checked implementation 
IE_ASSERT is implemented in such way 
https://github.com/marcinkusm/openvino/blob/master/src/inference/include/ie/ie_common.h#L439

which means that:  
 for debug build it will be regular assert  
for release it will be IE_THROW
so for now implementation is safe.

Reply:
@kbruniec based on my comment, can I leave IE_ASSERT or should i replace it with IE_EXCEPTION. IE_ASSERT itself will throw exception) for release build.